Utilities/bootstrap

```diff
@@ -584,17 +584,31 @@ def build_swift_driver(args):
         cmake_flags.append("-DCMAKE_OSX_DEPLOYMENT_TARGET=%s" % g_macos_deployment_target)
 
     build_with_cmake(args, cmake_flags, args.swift_driver_source_dir, args.swift_driver_build_dir)
-    
-def build_swift_crypto(args):
-    note("Building SwiftCrypto")
-    args.swift_crypto_build_dir = os.path.join(args.target_dir, "swift-crypto")
+
+def build_tsc(args):
+    note("Building TSC")
+    args.tsc_build_dir = os.path.join(args.target_dir, "tsc")
 
     cmake_flags = []
     if platform.system() == 'Darwin':
         cmake_flags.append("-DCMAKE_C_FLAGS=-target %s%s" % (get_build_target(args), g_macos_deployment_target))
         cmake_flags.append("-DCMAKE_OSX_DEPLOYMENT_TARGET=%s" % g_macos_deployment_target)
 
-    build_with_cmake(args, cmake_flags, args.swift_crypto_source_dir, args.swift_crypto_build_dir)
+    build_with_cmake(args, cmake_flags, args.tsc_source_dir, args.tsc_build_dir)
+
+def build_yams(args):
```

Comment:
There's a lot of duplication here — I realize that there was already some, but this is adding another case of it.  Would it make sense to consolidate some of this into a function?  I did that with the code to install a dylib and its associated module, and it might make sense to follow that model and pass parameters as needed.  The output directory could potentially be a return value.

Reply:
I agree. It makes sense to consolidate them into a function.

Reply:
I opened #3582 to consolidate the functions, instead of doing it in this PR, in order to avoid merge conflicts before this PR is ready.